### PR TITLE
testing/laszip: new aport

### PR DESCRIPTION
**Pull request by chambbj**
https://github.com/LASzip/LASzip
LASzip is a point cloud compression library.

### CODE REVIEW

**Comment by dawidd6** [comment]
I think your source should use github's release archive and not the commit one. Proper url to download 2.2.0 release is:
`https://github.com/LASzip/LASzip/archive/v2.2.0.tar.gz`

**Comment by jirutka** [comment]
Yes, @dawidd6 is right.

File: testing/laszip/APKBUILD [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+pkgrel=0
+pkgdesc="LASzip point cloud compression library"
+url="https://github.com/laszip/laszip"
+arch="all"
+license="LGPL2+"
+depends="libgcc libstdc++ musl"
+makedepends="cmake"
+install=""
+subpackages="$pkgname-dev"
+source="$pkgname-$pkgver.tar.gz::https://github.com/LASzip/$pkgname/archive/$_commitid.tar.gz"
```
Comment (dawidd6) [inline testing/laszip/APKBUILD]: Should be like this:
`source="$pkgname-$pkgver.tar.gz::https://github.com/LASzip/$pkgname/archive/v$pkgver.tar.gz"`

File: testing/laszip/APKBUILD [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Contributor: Bradley J Chambers <brad.chambers@gmail.com>
+# Maintainer: Bradley J Chambers <brad.chambers@gmail.com>
+pkgname=laszip
+pkgver=2.2.0
+_commitid=e7065cbc5bdbbe0c6e50c9d93d1cd346e9be6778
```
Comment (dawidd6) [inline testing/laszip/APKBUILD]: This is not needed

File: testing/laszip/APKBUILD [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+pkgdesc="LASzip point cloud compression library"
+url="https://github.com/laszip/laszip"
+arch="all"
+license="LGPL2+"
+depends="libgcc libstdc++ musl"
+makedepends="cmake"
+install=""
+subpackages="$pkgname-dev"
+source="$pkgname-$pkgver.tar.gz::https://github.com/LASzip/$pkgname/archive/$_commitid.tar.gz"
+builddir="$srcdir"/LASzip-$_commitid
```
Comment (dawidd6) [inline testing/laszip/APKBUILD]: `builddir="$srcdir"/LASzip-$pkgver`

or

`builddir="$srcdir"/LASzip-v$pkgver`

**Comment by chambbj** [comment]
@dawidd6 @jirutka Thank you for the quick review. v2.2.0 was released in 2013, with no further officially tagged releases. This repository is out of our control, but we pin to this particular commit for our own work, so I get how sticking with a `pkgver` of 2.2.0 can be misleading. Do you have other packages in a similar dilemma or any recommendations on how to handle the situation?

**Comment by dawidd6** [comment]
Maybe provide a patch with APKBUILD if it's not a big deal

**Comment by tmpfile** [comment]
You could also try asking the authors of LASzip to make a new release.
Either open an issue on their repo or send an email to the developers.

**Comment by chambbj** [comment]
FWIW, my slightly unconventional approach was following the example provided [here](https://github.com/alpinelinux/aports/blob/fb354bedba8f137e455e988f966a375cbad2ddbf/main/raspberrypi/APKBUILD), which I found linked from another issue IIRC.

I'm happy to try the patch approach too, had just started to implement that, although it's not really any more clear in terms of not lining up with a tagged release.

We already know that there is interest in releasing a new version in the very near future, so maybe this just waits until such time.

**Comment by dawidd6** [comment]
Yes in raspberrypi it is used, because there are not any releases on github page. Source is basically straight from master

**Comment by ncopa** [comment]
best is to ask upstream to tag a release instead of us doing it for them.

**Comment by chambbj** [comment]
Yes, I have a connection. This can hold until that release happens. Or if you'd rather, I can close and re-open at that time.

**Comment by ncopa** [comment]
We can either ship a `2.2.0` release as it was released, or we can wait til upstream make new release. We can not ship current git master or other random git commit and pretend it is `2.2.0` release.

Upstream issue: https://github.com/LASzip/LASzip/issues/13

**Comment by chambbj** [comment]
@ncopa Let's hold until next release.

**Comment by chambbj** [comment]
@ncopa We finally have a new release. This should be ready to go now.

**Pull request closed without merging**